src/main/java/uk/gov/cslearning/catalogue/api/CourseController.java

```diff
@@ -144,9 +150,80 @@ public CourseController(CourseRepository courseRepository, ResourceRepository re
     public ResponseEntity<Module> getModule(@PathVariable String courseId, @PathVariable String moduleId) {
         LOGGER.debug("Getting module {} of course {}", moduleId, courseId);
 
+        if (!courseRepository.existsById(courseId)) {
```

Comment:
module service will throw an exception if course is not found by `courseId`, so this is probably not needed - unless we want to change the response for "course not found"